bin/packages/get-babel-config.js

```diff
@@ -1,64 +1,38 @@
-/**
- * External dependencies
- */
-const { get, map } = require( 'lodash' );
-const babel = require( '@babel/core' );
-
-/**
- * WordPress dependencies
- */
-const { options: babelDefaultConfig } = babel.loadPartialConfig( {
-	configFile: '@wordpress/babel-preset-default',
-} );
-const { plugins, presets } = babelDefaultConfig;
-
-const overrideOptions = ( target, targetName, options ) => {
-	if ( get( target, [ 'file', 'request' ] ) === targetName ) {
-		return [ targetName, Object.assign(
-			{},
-			target.options,
-			options
-		) ];
+module.exports = function( environment = '', file ) {
+	/*
+	 * Specific options to be passed using the caller config option:
+	 * https://babeljs.io/docs/en/options#caller
+	 *
+	 * The caller options can only be 'boolean', 'string', or 'number' by design:
+	 * https://github.com/babel/babel/blob/bd0c62dc0c30cf16a4d4ef0ddf21d386f673815c/packages/babel-core/src/config/validation/option-assertions.js#L122
+	 */
+	const callerOpts = { caller: {
```

Comment:
Before this, I've tried to use an object shape like: 

```
{ caller: { 
    name: 'WP_BUILD'
    presetEnv: {
        // options to change would go here
    },
    pluginTransformRuntime: {
        // options to change would go here
    }
} }
```

This would be more flexible and future-proof. Could even be a basis for bringing extensibility for 3rd parties. We can't do that at the moment [by design](https://github.com/babel/babel/blob/bd0c62dc0c30cf16a4d4ef0ddf21d386f673815c/packages/babel-core/src/config/validation/option-assertions.js#L122) so added a note to keep an eye on that.